scripts/bert/baidu_ernie_data.py

```diff
@@ -0,0 +1,104 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=line-too-long
+"""GLUEBenchmark corpora."""
+
+__all__ = ['BaiduErnieXNLI']
+
+import os
+import sys
+import tarfile
+if sys.version_info[0] >= 3:
+    from urllib.request import urlretrieve
+else:
+    from urllib import urlretrieve
+
+
+from mxnet.gluon.utils import download, check_sha1, _get_repo_file_url
+
+from gluonnlp.data.dataset import TSVDataset
+from gluonnlp.data.registry import register
+from gluonnlp.base import get_home_dir
+
+_baidu_ernie_data_url = 'https://ernie.bj.bcebos.com/task_data.tgz'
+
+class _BaiduErnieDataset(TSVDataset):
+    def __init__(self, root, dataset_name, segment, **kwargs):
+        root = os.path.expanduser(root)
+        if not os.path.isdir(root):
+            os.makedirs(root)
+        self._root = root
+        download_data_path = os.path.join(root, 'task_data.tgz')
+        if not os.path.exists(download_data_path):
+            urlretrieve(_baidu_ernie_data_url, download_data_path)
+            tar_file = tarfile.open(download_data_path, mode='r:gz')
+            tar_file.extractall(root)
+        filename = os.path.join(self._root, 'task_data', dataset_name, '%s.tsv' % segment)
+        super(_BaiduErnieDataset, self).__init__(filename, **kwargs)
+
+
+@register(segment=['train', 'dev', 'test'])
+class BaiduErnieXNLI(_BaiduErnieDataset):
+    """ XNLI dataset
```

Comment:
The XNLI dataset released from Baidu <add link to Baidu's repo>

Reply:
added